app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -746,108 +738,103 @@ private ArtifactMetaDataDto createArtifactWithMetadataRaw(String groupId, String
+            // Delete labels
+            handle.createUpdate(sqlStatements.deleteLabels())
+                    .bind(0, normalizeGroupId(groupId))
+                    .bind(1, artifactId)
+                    .execute();
 
-                // Delete properties
-                handle.createUpdate(sqlStatements.deleteProperties())
-                        .bind(0, normalizeGroupId(groupId))
-                        .bind(1, artifactId)
-                        .execute();
+            // Delete properties
+            handle.createUpdate(sqlStatements.deleteProperties())
+                    .bind(0, normalizeGroupId(groupId))
+                    .bind(1, artifactId)
+                    .execute();
 
-                // Delete versions
-                handle.createUpdate(sqlStatements.deleteVersions())
-                        .bind(0, normalizeGroupId(groupId))
-                        .bind(1, artifactId)
-                        .execute();
+            deleteAllBranchesInArtifact(new GA(groupId, artifactId));
 
-                // Delete artifact rules
-                handle.createUpdate(sqlStatements.deleteArtifactRules())
-                        .bind(0, normalizeGroupId(groupId))
-                        .bind(1, artifactId)
-                        .execute();
+            // Delete versions
+            handle.createUpdate(sqlStatements.deleteVersions())
+                    .bind(0, normalizeGroupId(groupId))
+                    .bind(1, artifactId)
+                    .execute();
 
-                // Delete artifact row (should be just one)
-                int rowCount = handle.createUpdate(sqlStatements.deleteArtifact())
-                        .bind(0, normalizeGroupId(groupId))
-                        .bind(1, artifactId)
-                        .execute();
+            // Delete artifact rules
+            handle.createUpdate(sqlStatements.deleteArtifactRules())
+                    .bind(0, normalizeGroupId(groupId))
+                    .bind(1, artifactId)
+                    .execute();
+
+            // Delete artifact row (should be just one)
+            int rowCount = handle.createUpdate(sqlStatements.deleteArtifact())
+                    .bind(0, normalizeGroupId(groupId))
+                    .bind(1, artifactId)
+                    .execute();
+
+            if (rowCount == 0) {
+                throw new ArtifactNotFoundException(groupId, artifactId);
+            }
 
-                if (rowCount == 0) {
-                    throw new ArtifactNotFoundException(groupId, artifactId);
-                }
-                return versions;
-            });
             deleteAllOrphanedContent();
-            return res;
-        } catch (ArtifactNotFoundException e) {
-            throw e;
-        } catch (Exception e) {
-            throw new RegistryStorageException(e);
-        }
+
+            return versions;
+        });
     }
 
 
     @Override
     @Transactional
```

Comment:
Are these annotations still necessary?  I seem to recall you've changed how this works so that the Tx boundary happens in the handle, no?

Reply:
Not in this PR, I didn't want to do too much at once.

Reply:
So you somehow figured out how to get around the rollback issue?  Or that's still an issue and a fix is coming in a future PR?

Reply:
The issue is still present, it just shows up under specific circumstances. I'll create a reproducer test in the upcoming PR.